src/compiler/checker.ts

```diff
@@ -23935,9 +23935,21 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     function inferReverseMappedType(sourceType: Type, target: MappedType, constraint: IndexType): Type {
         const typeParameter = getIndexedAccessType(constraint.type, getTypeParameterFromMappedType(target)) as TypeParameter;
         const templateType = getTemplateTypeFromMappedType(target);
-        const inference = createInferenceInfo(typeParameter);
-        inferTypes([inference], sourceType, templateType);
-        return getTypeFromInference(inference) || unknownType;
+        const inferences = [createInferenceInfo(typeParameter)];
+        inferTypes(inferences, sourceType, templateType);
+        const wholeInferredType = getTypeFromInference(inferences[0]);
+        if (!wholeInferredType && inferences.length > 1) {

```

Comment:
If any direct inference was made for `T[K]` then we will ignore inferences made for its properties. I think it's the right thing to do here to preserve backward compatibility. This means that, In a sense, the property-based inferences simply~ have a lower inference priority.